### PR TITLE
Add support for custom logback.xml and tracing.xml

### PR DESCRIPTION
**Pull request by Donnerbart**
https://hivemq.kanbanize.com/ctrl_board/22/cards/24584/details/

### CODE REVIEW

**Review by afalhambra-hivemq — APPROVED** [review]

File: charts/hivemq-platform/templates/hivemq-configuration.yml [hunk 1]
```diff
@@ -357,5 +361,9 @@ data:
         </publish-sampling>
       </sampling>
     </tracing>
-    {{- end -}}
+  {{- end }}
+  {{- if .Values.config.customLogbackConfig }}
```
[the remark this answers is not in the document]
Comment (Donnerbart) [reply charts/hivemq-platform/templates/hivemq-configuration.yml:365]: For the `customTracingConfig` this worked, but if I add the `-` here, my test is starting to fail:
```
 FAIL  test HiveMQ Platform configmap   charts/hivemq-platform/tests/hivemq_configuration_test.yaml
        - with custom logback.xml

                - asserts[2] `exists` fail
                        Template:       hivemq-platform/templates/hivemq-configuration.yml
                        DocumentIndex:  0
                        Path:   data["logback.xml"] expected to exists

                - asserts[3] `equal` fail
                        Template:       hivemq-platform/templates/hivemq-configuration.yml
                        DocumentIndex:  0
                        Error:
                                unknown path data["logback.xml"]

```
I cannot figure this one out :(

File: charts/hivemq-platform/tests/hivemq_configuration_test.yaml [hunk 2]
```diff
@@ -53,6 +53,10 @@ tests:
                 </listeners>
               </control-center>
             </hivemq>
+      - exists:
+          path: data["tracing.xml"]
```
[the remark this answers is not in the document]
Comment (Donnerbart) [reply charts/hivemq-platform/tests/hivemq_configuration_test.yaml:57]: Good one!

File: charts/hivemq-platform/values.yaml [hunk 3]
```diff
@@ -322,11 +323,16 @@ config:
...
   #          Please, consider using `additionalInitContainers` value instead.
   overrideInitContainers: ""
-
   # Enables HiveMQ Data Hub
   dataHub:
     dataValidationEnabled: false
     behaviorValidationEnabled: false
+  # Inlines a custom HiveMQ Platform tracing configuration (tracing.xml) from a file:
+  # --set-file config.customTracingXml=your-logback.xml
+  customTracingConfig: ""
```
[the remark this answers is not in the document]
Comment (Donnerbart) [reply charts/hivemq-platform/values.yaml:332]: My reasoning was to align it with `overrideHiveMQConfig` (which is not `overrideHiveMQXml`). As you can see I also wasn't sure which variant is better though.

As discussed on Discord, please sync with Ryan what is the better naming.

File: charts/hivemq-platform/values.yaml [hunk 4]
```diff
@@ -322,11 +323,16 @@ config:
...
+  # Inlines a custom HiveMQ Platform logging configuration (logback.xml) from a file:
+  # --set-file config.customLogbackXml=your-logback.xml
+  customLogbackConfig: ""
```
[the remark this answers is not in the document]
Comment (Donnerbart) [reply charts/hivemq-platform/values.yaml:335]: I would agree if we want the option to make the `tracing.xml` and `logback.xml` configurable in the Helm chart itself (like break down into multiple options). If that is not intended, I would keep it simple like this.

As discussed on Discord, please sync with Ryan how we should proceed.

**Pull request merged**